### PR TITLE
feat: Copy to clipboard all text nodes as text

### PR DESCRIPTION
**Pull request by Fausto95**
Kapture 2022-04-05 at 14 56 25

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -5483,6 +5484,10 @@ class App extends React.Component<AppProps, AppState> {
     if (probablySupportsClipboardWriteText && elements.length > 0) {
       options.push(actionCopyAsSvg);
     }
+
+    if (probablySupportsClipboardWriteText && elements.length > 0) {
+      options.push(copyAllTextNodesAsText);
+    }
```
[the remark this answers is not in the document]
Comment (dwelle) [reply src/components/App.tsx:5490]: I'd show for any selection that contains at least 1 text element.

[the remark this answers is not in the document]
Comment (Fausto95) [reply src/components/App.tsx]: I believe that is better to show it only when all the elements are text nodes

**Review by ad1992 — APPROVED** [review]
looks good, thanks @Fausto95 ❤️

**Comment by dwelle** [comment]
Actually, this isn't working quite right.

1. You should be able to copy text of selected elements even if there are other non-text elements on the scene
2. It should only copy selected elements, not all elements on canvas
3. The contextMenu action should only work on selection (elements), not on canvas

Haven't reviewed the code itself.

**Comment by dwelle** [comment]
I'd also put `\n\n` between texts of each element, i.e. make each element into a paragraph.

**Comment by Fausto95** [comment]
> Actually, this isn't working quite right.
> 
> 1. You should be able to copy text of selected elements even if there are other non-text elements on the scene
> 2. It should only copy selected elements, not all elements on canvas
> 3. The contextMenu action should only work on selection (elements), not on canvas
> 
> Haven't reviewed the code itself.

Oh yea, you're right. I'll make a follow up PR

File: src/locales/en.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "copy": "Copy",
     "copyAsPng": "Copy to clipboard as PNG",
     "copyAsSvg": "Copy to clipboard as SVG",
+    "copyAllTextNodesAsText": "Copy to clipboard as a single text element",
```
Comment (dwelle) [inline src/locales/en.json:12]: `as single text element` doesn't make much sense (element). I'd change it to `Copy to clipboard as text`. Not sure if that much clearer though.

**Pull request merged**